pkg/pillar/cmd/downloader/syncop.go

```diff
@@ -288,9 +289,21 @@ func handleSyncOp(ctx *downloaderContext, key string,
 			}
 			continue
 		}
+
+		// Finalize: fsync and atomically move temp -> final.
+		if f, ferr := os.OpenFile(tempLocFilename, os.O_RDWR, 0); ferr == nil {
```

Comment:
If this Open fails, shouldn't you return an error instead of trying the Rename below and have that fail?

Reply:
I added the code, but I think it is not necessary as the rename will fail below.